### ISSUE TITLE
[java] LawOfDemeter: False positive with 'this' pointer

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2174]
https://github.com/pmd/pmd/issues/2160#issuecomment-565639733

**Affects PMD Version:** 
6.20.0

**Rule:**
[LawOfDemeter](https://pmd.github.io/latest/pmd_rules_java_design.html#lawofdemeter)

**Description:**
LawOfDemeter does not recognise 'this' pointer and includes its dot accessor as a count towards method chaining. It must not. Similarly for 'super' while calling parent class functions. 

**Code Sample demonstrating the issue:**

```java
package pmdtests;

public class Foo {
  private final int i;

  public Foo(int i) {
    this.i = i;
  }


  public void meth() {
    System.out.println(this.i); // wrong violation: method chain calls
    System.out.println(super.i); // wrong violation: method chain calls
    System.out.println(i);      // no violation
  }

  public void method(Foo this) {
    System.out.println(this.i);  // wrong violation: method chain calls
  }

  public void method(Foo this, String... args) {
    System.out.println(this.i);  // wrong violation: method chain calls
    for (String arg: args)
      System.out.println(arg);
  }
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
